include/circt-c/Dialect/HW.h

```diff
@@ -88,6 +91,10 @@ MLIR_CAPI_EXPORTED MlirType hwStructTypeGet(MlirContext ctx,
 MLIR_CAPI_EXPORTED MlirType hwStructTypeGetField(MlirType structType,
                                                  MlirStringRef fieldName);
 
+MLIR_CAPI_EXPORTED MlirType hwIntTypeGet(MlirIdentifier parameter);
```

Comment:
Can't param int types contain arbitrary param expressions? If so, make this accept an `MlirAttribute` and pass in the `ParamDeclRefAttr` to get the same behavior.